testsuite/unittests/test_symmetric_connections.sli

```diff
@@ -0,0 +1,155 @@
+/*
+ *  test_symmetric_connections.sli
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/* BeginDocumentation
+
+Name: testsuite::test_symmetric_connections - Tests that the functionality to create
+symmetric connections works properly.
+
+Synopsis: (test_symmetric_connections) run -> NEST exits if test fails
+
+Description: 
+This test ensures that the functionality to create symmetric connections
+works properly.
+
+In more detail the test ensures that
+- the symmetric flag works properly with one-to-one connection rule
+- the usage of the symmetric flag with any other connections throws
+  an NotImplemented exception
+
+Author: Jan Hahne, 2016-04-22
+*/
+
+(unittest) run
+/unittest using
+
+M_ERROR setverbosity
+
+ResetKernel
+
+/iaf_neuron 10 Create
+
+/set1 1 5 cvgidcollection def
+/set2 6 10 cvgidcollection def
```

Comment:
@janhahne I noticed that here an in several other places below you have code outside any `[pass|assert]_or_die` blocks. This is not really optimal, because it creates dependencies between parts of the code. A better way is to create a little helper function (e.g., `/setup`, `/prepare`, or so) that sets up the necessary pieces and the call that from each of the test blocks.